packages/cli/src/lib/mini-oxygen/workerd-inspector-proxy.ts

```diff
@@ -0,0 +1,160 @@
+import crypto from 'node:crypto';
+import {
+  createServer,
+  type IncomingMessage,
+  type ServerResponse,
+} from 'node:http';
+import {WebSocketServer, type WebSocket, type MessageEvent} from 'ws';
+import {type Protocol} from 'devtools-protocol';
+import {type InspectorWebSocketTarget} from './workerd-inspector.js';
+
+export function createInspectorProxy(
+  port: number,
+  inspectorConnection?: {ws?: WebSocket},
+) {
+  /**
+   * A unique identifier for this debugging session.
+   */
+  const sessionId = crypto.randomUUID();
+  /**
+   * WebSocket connection to the local debugger (e.g. VSCode, DevTools).
+   */
+  let debuggerWs: WebSocket | undefined = undefined;
+  /**
+   * WebSocket connection to the Workerd inspector.
+   */
+  let inspectorWs: WebSocket | undefined = inspectorConnection?.ws;
+
+  const server = createServer((req: IncomingMessage, res: ServerResponse) => {
+    // Remove query params. E.g. `/json/list?for_tab`
+    const url = req.url?.split('?')[0] ?? '';
+
+    switch (url) {
+      // We implement a couple of well known end points
```

Comment:
I'm curious how you knew how to do this?

Reply:
Lots of this comes from Wrangler itself!